app/code/community/Bolt/Boltpay/Model/Order.php

```diff
@@ -114,26 +113,27 @@ public function createOrder($reference, $sessionQuoteId = null, $isAjaxRequest =
             ///  the Bolt transaction.
             //////////////////////////////////////////////////////////////////////////////////
             $shippingMethodCode = null;
-            $shippingAndTaxModel = Mage::getModel("boltpay/shippingAndTax");
             if ($transaction->order->cart->shipments) {
-
+                /** @var Bolt_Boltpay_Model_ShippingAndTax $shippingAndTaxModel */
+                $shippingAndTaxModel = Mage::getModel("boltpay/shippingAndTax");
                 $shippingAndTaxModel->applyShippingAddressToQuote($immutableQuote, $transaction->order->cart->shipments[0]->shipping_address);
                 $shippingMethodCode = $transaction->order->cart->shipments[0]->reference;
             }
 
             $immutableQuote->getShippingAddress()->setShouldIgnoreValidation(true)->save();
             $immutableQuote->getBillingAddress()->setShouldIgnoreValidation(true)->save();
 
+            $shippingAddress = $immutableQuote->getShippingAddress();
+            $shippingAddress->setCollectShippingRates(true)->collectShippingRates();
```

Comment:
@YevhenShBolt this is the only potential issue I see. I don't think we need to collect shipping rates if they already exist since the shipping rate will be collected during the shipping step. I think we only recollect the shipping rates for legacy purposes. Can you confirm? Everything else looks good to me.

Reply:
I did not catch such issue earlier.
I think we would see before such a problem if it existed, this part of the code there ever since:
`390a6d87 John McCombs <john.mccombs@bolt.com> on 2018-09-25 at 13:57`